android/app/src/main/java/org/unicef/ecar/bebbo/MainActivity.java

```diff
@@ -6,108 +6,76 @@
 import android.util.Log;
 
 import com.facebook.react.ReactActivity;
+import com.facebook.react.ReactActivityDelegate;
+import com.facebook.react.defaults.DefaultNewArchitectureEntryPoint;
+import com.facebook.react.defaults.DefaultReactActivityDelegate;
 import com.google.android.gms.common.GoogleApiAvailability;
 import com.google.android.gms.security.ProviderInstaller;
 // import org.devio.rn.splashscreen.SplashScreen;
-import org.devio.rn.splashscreen.SplashScreen; 
-public class MainActivity extends ReactActivity implements ProviderInstaller.ProviderInstallListener {
-
-    private static final int ERROR_DIALOG_REQUEST_CODE = 1;
-
-    private boolean retryProviderInstall;
+import org.devio.rn.splashscreen.SplashScreen;
+public class MainActivity extends ReactActivity implements ProviderInstaller.ProviderInstallListener{
 
   /**
    * Returns the name of the main component registered from JavaScript. This is used to schedule
    * rendering of the component.
    */
-//Update the security provider when the activity is created.
-   @Override
-    protected void onCreate(Bundle savedInstanceState) {
-       Log.d("Mainactivity", "Mainactivity java oncreate helloo");
-        SplashScreen.show(this,R.style.SplashScreenTheme,R.id.lottie);  // here
-        SplashScreen.setAnimationFinished(true);
-        super.onCreate(savedInstanceState);
-       Intent intent = getIntent();
-       String action = intent.getAction();
-       Uri data = intent.getData();
-       ProviderInstaller.installIfNeededAsync(this, this);
-    }
-    @Override
-    protected void onSaveInstanceState(Bundle outState) {
-        super.onSaveInstanceState(outState);
-        if (outState != null) { outState.clear(); }
-    }
+  private static final int ERROR_DIALOG_REQUEST_CODE = 1;
+
+  private boolean retryProviderInstall;
+
+  //Update the security provider when the activity is created.
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    Log.d("Mainactivity", "Mainactivity java oncreate helloo");
+    SplashScreen.show(this,R.style.SplashScreenTheme,R.id.lottie);  // here
+    SplashScreen.setAnimationFinished(true);
+    super.onCreate(savedInstanceState);
+    Intent intent = getIntent();
+    String action = intent.getAction();
+    Uri data = intent.getData();
+    ProviderInstaller.installIfNeededAsync(this, this);
+  }
+  @Override
+  protected void onSaveInstanceState(Bundle outState) {
+    super.onSaveInstanceState(outState);
+    if (outState != null) { outState.clear(); }
+  }
   @Override
   protected String getMainComponentName() {
     return "ParentBuddyApp";
   }
+  @Override
+  public void onProviderInstalled() {
+    // Provider is up-to-date, app can make secure network calls.
+  }
 
-    /**
-     * This method is only called if the provider is successfully updated
-     * (or is already up-to-date).
-     */
```

Comment:
This comment looks useful. Let's keep it.

Reply:
okay sure @Akhror